--- a/src/makefile.mingw
+++ b/src/makefile.mingw
@@ -1,7 +1,64 @@
 # Copyright (c) 2009-2010 Satoshi Nakamoto
+# Copyright (c) 2012 The PPCoin developers
 # Distributed under the MIT/X11 software license, see the accompanying
 # file license.txt or http://www.opensource.org/licenses/mit-license.php.
 
+# Windows commandline build procedure:
+# - Install MinGW following http://www.mingw.org/wiki/Getting_Started.
+#   Install with the C++ and MSYS options checked
+#
+# - Add/edit 'MAKE' environment variable with value '/c/MinGW32/bin/mingw32-make.exe'
+#
+# - Build openssl library version: 1.0.1b
+#   download from  http://www.openssl.org/source/
+#   Extract to c:\openssl-1.0.1b-mgw
+#   In MinGW MSYS:
+#     ./config
+#     make
+#
+# - Build Berkeley DB library version: 4.8.30.NC
+#   download from  http://www.oracle.com/technology/software/products/berkeley-db/index.html
+#   Extract to c:\db-4.8.30.NC-mgw
+#   In MinGW MSYS:
+#     cd build_unix
+#     sh ../dist/configure --disable-replication --enable-mingw --enable-cxx --prefix=/usr/local
+#   Edit db.h@113 in build_unix
+#   from
+#     typedef pthread_t db_threadid_t;
+#   to 
+#     typedef u_int32_t db_threadid_t;
+#   Then
+#     make
+#
+# - Build Boost C++ library version: 1.47.0
+#   download from http://www.boost.org/users/download/
+#   Extract to c:\boost-1.47.0-mgw
+#   Install Boost.Build:
+#     cd tools\build\v2
+#     bootstrap.bat
+#     b2 install --prefix=BOOST_BUILD_INSTALL_DIR
+#   Add BOOST_BUILD_INSTALL_DIR to your PATH system environment variable
+#   Build boost library in MSDOS:
+#     cd c:\boost-1.47.0-mgw
+#     bjam toolset=gcc --build-type=complete stage
+#
+# - Build Miniupnp library version: 1.6
+#   download from http://miniupnp.free.fr/files/
+#   file name is miniupnpc-1.6.20120509.tar.gz
+#   Extract to c:\miniupnpc-1.6-mgw
+#   In MinGW MSYS shell:
+#     cd /c/miniupnpc-1.6-mgw
+#     make -f Makefile.mingw
+#     mkdir miniupnpc
+#     cp *.h miniupnpc/
+#
+# - Build ppcoind.exe
+#   in MinGW MSYS
+#     cd ppcoin/src
+#     make ppcoind.exe -f makefile.mingw USE_UPNP=1
+#
+#
+
 USE_UPNP:=0
 
 INCLUDEPATHS= \
@@ -15,10 +72,10 @@ LIBPATHS= \
  -L"C:\openssl-1.0.1b-mgw"
 
 LIBS= \
- -l boost_system-mgw45-mt-s-1_47 \
- -l boost_filesystem-mgw45-mt-s-1_47 \
- -l boost_program_options-mgw45-mt-s-1_47 \
- -l boost_thread-mgw45-mt-s-1_47 \
+ -l boost_system-mgw46-mt-s-1_47 \
+ -l boost_filesystem-mgw46-mt-s-1_47 \
+ -l boost_program_options-mgw46-mt-s-1_47 \
+ -l boost_thread-mgw46-mt-s-1_47 \
  -l db_cxx \
  -l ssl \
  -l crypto
@@ -61,15 +118,16 @@ OBJS= \
     obj/util.o \
     obj/wallet.o \
     obj/walletdb.o \
-    obj/noui.o
+    obj/noui.o \
+    obj/kernel.o
 
 
-all: bitcoind.exe
+all: ppcoind.exe
 
 obj/%.o: %.cpp $(HEADERS)
 	g++ -c $(CFLAGS) -o $@ $<
 
-bitcoind.exe: $(OBJS:obj/%=obj/%)
+ppcoind.exe: $(OBJS:obj/%=obj/%)
 	g++ $(CFLAGS) -o $@ $(LIBPATHS) $^ $(LIBS)
 
 TESTOBJS := $(patsubst test/%.cpp,obj-test/%.o,$(wildcard test/*.cpp))
@@ -81,7 +139,7 @@ test_bitcoin.exe: $(TESTOBJS) $(filter-out obj/init.o,$(OBJS:obj/%=obj/%))
 	g++ $(CFLAGS) -o $@ $(LIBPATHS) $^ -lboost_unit_test_framework $(LIBS)
 
 clean:
-	-del /Q bitcoind test_bitcoin
+	-del /Q ppcoind test_bitcoin
 	-del /Q obj\*
 	-del /Q obj-test\*
 	-del /Q build.h
diff --git a/src/makefile.osx b/src/makefile.osx
index 11c6248..c0883a9 100644
