--- a/src/qt/forms/sendcoinsentry.ui
+++ b/src/qt/forms/sendcoinsentry.ui
@@ -90,7 +90,7 @@
      <item>
       <widget class="QValidatedLineEdit" name="payTo">
        <property name="toolTip">
-        <string>The address to send the payment to  (e.g. 1NS17iag9jJgTHD1VXjvLCEnZuQ3rJDE9L)</string>
+        <string>The address to send the payment to</string>
        </property>
        <property name="maxLength">
         <number>34</number>
diff --git a/src/qt/guiutil.cpp b/src/qt/guiutil.cpp
index f1e8a5f..166c02c 100644
