--- a/src/qt/walletmodel.h
+++ b/src/qt/walletmodel.h
@@ -50,6 +50,7 @@ public:
     TransactionTableModel *getTransactionTableModel();
 
     qint64 getBalance() const;
+    qint64 getStake() const;
     qint64 getUnconfirmedBalance() const;
     int getNumTransactions() const;
     EncryptionStatus getEncryptionStatus() const;
@@ -120,7 +121,7 @@ private:
 
 signals:
     // Signal that balance in wallet changed
-    void balanceChanged(qint64 balance, qint64 unconfirmedBalance);
+    void balanceChanged(qint64 balance, qint64 stake, qint64 unconfirmedBalance);
 
     // Number of transactions in wallet changed
     void numTransactionsChanged(int count);
diff --git a/src/rpcdump.cpp b/src/rpcdump.cpp
index 2db4882..89e0405 100644
